### DISCUSSION TITLE
Nuxt 3: When is a store considered to be "outside of `setup()`" and I have to use `$pinia`?

### DISCUSSION THREAD

**Discussion by oemer-aran** [vuejs/pinia#2299] [Help and Questions]
The docs do not answer this clearly: https://pinia.vuejs.org/ssr/nuxt.html#using-the-store-outside-of-setup
So here are the cases, when I don't know if I have to use `$pinia` from the nuxt context:

- Inside a route middleware
- Inside a server middleware
- Inside a universal plugin (client+server)
- Inside a server plugin
- Inside a client plugin
- Inside another store (The store can be used inside a setup, but also inside middlewares, plugins etc...)
- Inside a composable (The composable can be used inside a setup, but also inside middlewares, plugins etc...)

Related question:
- What are the drawbacks if you use `$pinia` when you shouldn't?
  - If there is no drawback, why not always use `$pinia` to be always "safe"?

Ultimately, this might be just a request to improve the docs.

**Comment by mgreer** [comment]
Same question. It is unclear what "outside" means, and why it needs `$pinia`.

**Comment by oemer-aran** [comment]
Any insides? Maybe @danielroe

**Comment by Me-Phew** [comment]
Also do you know how to actually use $pinia inside route middleware?

**Reply by oemer-aran** [reply]
Did you try `const { $pinia }  = useNuxtApp()`?

**Reply by Me-Phew** [reply]
Yes I was wondering if this is the right way to do it.

**Reply by oemer-aran** [reply]
I am not even sure if you need it there. I don't use it in route middleware and I don't see any downsides so far. Thats where all my confusion comes from with `$pinia`. Don't know where to use it.

**Reply by Me-Phew** [reply]
Alright thanks for help
